### ISSUE TITLE
It takes me about 10s to load an article list of nearly 30k documents

### ISSUE THREAD

**Issue by Stupidism** [payloadcms/payload#2227]
It takes me about 10s to load the article list 

![image](https://user-images.githubusercontent.com/5801015/222611844-046094d6-4b48-4db5-8522-685633483bb7.png)

I inspected the slow API. If I remove the draft option. I will be very fast.

```
https://mysite.com/api/articles?locale=en&depth=0&draft=true&limit=10
```

vs

```
https://mysite.com/api/articles?locale=en&depth=0&limit=10
```

And it cannot reproduce in my local machine. 

And I ran the migration script as the document said. https://github.com/payloadcms/payload/blob/master/CHANGELOG.md#-versions-may-need-to-be-migrated

_Originally posted by @Stupidism in https://github.com/payloadcms/payload/discussions/2217_

**Comment by jmikrut** [comment]
Hey there – that is certainly not acceptable. Do you have indexes created on the ‘updatedAt’ and ‘parent’ fields on your versions collection?

We will look into this tomorrow 👍

**Comment by JarrodMFlesch** [comment]
@Stupidism, might want to trim the top bit of that first image 😅

**Comment by jmikrut** [comment]
Hmm, this is very strange. 

I have absolutely no idea why your GraphQL is functioning as expected but REST is not. Are you able to use a tool like MongoDB Compass to connect to your database? If so, you should try and recreate the aggregate that you are running within MongoDB Compass directly.

This also might be to do with AWS' DocumentDB. I believe that Mongo will fall back on writing to disk if it does not have an index available, or if the resulting data set is too large for it to be executed in memory. That could be why you are seeing COLLSCAN. This is definitely related to COLLSCAN—the question is - why? 

I just seeded 40K documents and it still loads blazingly fast for me. Here is a screenshot of my working database's indexes within MongoDB Compass:

![Screen Shot 2023-03-06 at 11 34 08 AM](https://user-images.githubusercontent.com/6651896/223173784-969ebf6b-aa61-4003-88ad-eb0ff400d12e.png)

**Comment by jmikrut** [comment]
Oops - did not mean to close. I didn't even know I did! 

Question - is there a way that you could clone your exact database, run on Atlas (or a local MongoDB Community), and then re-assess the speed here?

**Comment by JarrodMFlesch** [comment]
@Stupidism The versions screen crashing was a [reported issue](https://github.com/payloadcms/payload/issues/2228) and has since been fixed, just letting you know!

**Comment by jmikrut** [comment]
Hey @Stupidism — I think this is definitely the move. 

> For example, if we set a flag in versions table latest: true?

The team and I have been discussing a way to do exactly this, and I think this will circumvent your issue.

Give us a bit here and we will get a plan in place to make this or something similar to this happen.

**Comment by jmikrut** [comment]
Hey @Stupidism — what version of local Mongo are you using? Do you have it installed locally on your computer, or are you using Docker or similar? 

It's very strange that we can't reproduce this at all. We're looking into fixes still, but I really need to be able to reproduce your exact scenario before being confident in a fix. Still digging, but the more info you can provide about your local MongoDB, the better.

We've also recreated your exact aggregate pipeline directly in Mongo Compass, and it's still quite fast:

![image](https://user-images.githubusercontent.com/6651896/224758092-4bc1242d-2d42-4ec1-abed-1c89151121f5.png)

**Comment by jmikrut** [comment]
Hmm, ok - so just so I am clear, on local, you are actually seeing pretty good execution time - 785ms, right? Did you change anything? Before I believe you mentioned 5s+ even trying this locally. But this looks about similar to what I am seeing on my end when I test locally. Good that it's using the `updatedAt` index.

Let me know and we can go from there. Thanks!!

**Comment by davidko604** [comment]
> Hi @jmikrut, This is the url you can compare the speed:
> 
> https://seo-payload.stage.getjerry.com/api/articles?locale=en&depth=0&draft=true&page=1
> 
> https://seo-payload.stage.getjerry.com/api/articles?locale=en&depth=0&page=1
> 
> ```shell
> curl -o /dev/null -s -w 'Establish Connection: %{time_connect}s\nTTFB: %{time_starttransfer}s\nTotal: %{time_total}s\n' https://seo-payload.stage.getjerry.com/api/articles\?locale\=en\&depth\=0\&page\=1
> Establish Connection: 0.000000s
> TTFB: 0.000000s
> Total: 5.009390s
> 
> curl -o /dev/null -s -w 'Establish Connection: %{time_connect}s\nTTFB: %{time_starttransfer}s\nTotal: %{time_total}s\n'  https://seo-payload.stage.getjerry.com/api/articles\?locale\=en\&depth\=0\&draft\=true\&page\=1
> Establish Connection: 0.007475s
> TTFB: 21.427764s
> Total: 21.427931s
> ```

Hi @jmikrut ,

My team is also having similar issues with one of our collections. With drafts disabled, we get very quick collection list load time but with drafts enabled we are having long load times.

```look at args.operation to make sure it's the read operation
look at args.req.origin to see if the request is coming from the admin UI
look at args.id to see if there is an ID (AKA this is the find and not findByID operation). You should only disable draft=true if it's on the find operation itself```

This works, but it looks like a brittle solution for us and are concerned that with a payload update our experience can be broken at some point. If you use a similar data structure to above are you able to mock or reproduce this issue?

**Comment by klapec** [comment]
Has anyone tested whether https://github.com/payloadcms/payload/releases/tag/v1.15.3 improves query speeds for draft-enabled collections on DocumentDB?

**Comment by JarrodMFlesch** [comment]
> Has anyone tested whether https://github.com/payloadcms/payload/releases/tag/v1.15.3 improves query speeds for draft-enabled collections on DocumentDB?

It should have the same affect on DocumentDB, since it removes the need for the aggregation step when querying drafts!

**Comment by klapec** [comment]
Sounds promising! Will check it tomorrow, I'll let you know.

**Comment by jmikrut** [comment]
Yes, this issue has been a thorn in our side for quite some time now but I am pretty confident that this new approach will 100% solve this issue for all Mongo-like databases from here on out.

It will be defaulted in 2.0, and we wanted to get it out now as a feature flag so you guys don't have to wait 👍 thanks to Jarrod for the quick moves here!
